plugin/storage/integration/integration_test.go

```diff
@@ -0,0 +1,297 @@
+	}
+
+	if !assert.True(t, found) {
+		t.Log("\t Expected:", expected)
+		t.Log("\t Actual  :", actual)
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestGetOperations(t *testing.T) {
+	t.Log("Testing GetOperations ...")
+
+	expected := []string{"operation-list-test1", "operation-list-test2", "operation-list-test3"}
+	traces, err := getTraceFixtures()
+	require.NoError(t, err)
+	require.NoError(t, s.writeTraces(traces))
+
+	s.refresh()
+
+	var found bool
+	var actual []string
+	for i := 0; i < 30; i++ {
+		s.logger.Info(fmt.Sprintf("Waiting for storage backend to update documents, iteration %d out of %d", i+1, 30))
+		actual, err = s.reader.GetOperations("query05-service")
+		require.NoError(t, err)
+		if found = assert.ObjectsAreEqualValues(expected, actual); found {
+			break
+		}
+		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
+	}
+
+	if !assert.True(t, found) {
+		t.Log("\t Expected:", expected)
+		t.Log("\t Actual  :", actual)
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestGetTrace(t *testing.T) {
+	t.Log("Testing GetTrace ...")
+
+	expected, err := getTraceFixture(1)
+	require.NoError(t, err)
+	require.NoError(t, s.writeTrace(expected))
+	traceID := model.TraceID{Low: 1}
+
+	s.refresh()
+
+	var found bool
+	var actual *model.Trace
+	for i := 0; i < 30; i++ {
+		s.logger.Info(fmt.Sprintf("Waiting for storage backend to update documents, iteration %d out of %d", i+1, 30))
+		actual, err = s.reader.GetTrace(traceID)
+		if err == nil {
+			found = len(actual.Spans) == len(expected.Spans)
+			if found {
+				CompareTraces(t, expected, actual)
+				break
+			}
+		}
+		time.Sleep(100 * time.Millisecond) // Will wait up to 3 seconds at worst.
+	}
+
+	if !assert.True(t, found) {
+		CompareTraces(t, expected, actual)
+	}
+	assert.NoError(t, s.cleanUp())
+}
+
+func (s *StorageIntegration) IntegrationTestFindTraces(t *testing.T) {
+	t.Log("Testing FindTraces ...")
+	t.Log("\t(Note: all cases include ServiceName + StartTime range)")
+	queries, err := getQueries()
+	require.NoError(t, err)
+	for _, query := range queries {
+		t.Logf("\t\t* Query case: + %s", query.Caption)
+		s.integrationTestFindTracesByQuery(t, query.Query, query.ExpectedFixtures)
+	}
+}
+
+func (s *StorageIntegration) integrationTestFindTracesByQuery(t *testing.T, query *spanstore.TraceQueryParameters, expectedTraces []int) {
+	traces, err := getTraceFixtures()
+	require.NoError(t, err)
+	expected := getSubsetOfTraces(traces, expectedTraces)
+	require.NoError(t, s.writeTraces(expected))
+	require.NoError(t, s.refresh())
+
+	var found bool
+	var actual []*model.Trace
+	for i := 0; i < 30; i++ {
+		s.logger.Info(fmt.Sprintf("Waiting for storage backend to update documents, iteration %d out of %d", i+1, 30))
+		actual, err = s.reader.FindTraces(query)
+		if err == nil {
+			if len(actual) == query.NumTraces {
```

Comment:
the thing is, this might not really be a test case anyway, since we changed to one-trace-per-query integ testing, so i'm not actually ever hitting this case